### PR TITLE
AddedChanges

### PR DESCRIPTION
**Pull request by dineshdeverakonda**
Created a URL field on Case Collaboration called Case Link and populate this field in the trigger based on if the User selected in the lookup is an Agent/Customer and use this field in the View Case button in the email template(This way we can avoid creating new template).
CHANGES
1) Changes in TriggerHelperClass and Trigger.
2)Made changes in Email Templates added view button on all collaboration email templates(english french spanish)
3)Created a custom field on CaseCollaborator called CaseLink__c

### CODE REVIEW

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationTriggerHelper.cls [hunk 1]
```diff
@@ -55,7 +55,47 @@ public class DEL_CaseCollaborationTriggerHelper {
...
+     /**
+    * @ author      : Dinesh Chandra
+    * @ description : This method handles creation of URL's for Community Portal for Customers and Salesforce Login for Agents.
+    * @ params      : list_CaseCollaborators - list of Case collaborator records passed from trigger
+    **/
+    public static void populateFields(List<DEL_CaseCollaborator__c> list_CaseCollaborators) {
+        String strCommnunityURL = '';
+        objEmailAlertConfiguration = DEL_EmailAlertConfiguration__c.getOrgDefaults();
+        String strSFURL = '';
+        String strHTML = '';
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationTriggerHelper.cls:67]: remove this as it's not being used

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationTriggerHelper.cls [hunk 2]
```diff
@@ -55,7 +55,47 @@ public class DEL_CaseCollaborationTriggerHelper {
...
+        strSFURL = URL.getSalesforceBaseUrl().toExternalForm();
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationTriggerHelper.cls:68]: You can assign this in line 66 itself

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationTriggerHelper.cls [hunk 3]
```diff
@@ -55,7 +55,47 @@ public class DEL_CaseCollaborationTriggerHelper {
...
+        SiteDetail objSiteDetail = [SELECT
+                                    SecureURL 
+                                    FROM SiteDetail
+                                    WHERE DurableId = :objSite.Id
+                                    ];
+        strCommnunityURL = objSiteDetail.SecureURL;
+        Set<Id> set_UserIds = new Set<Id>();
+        for (DEL_CaseCollaborator__c objCaseCollaborator : list_CaseCollaborators) {
+            set_UserIds.add(objCaseCollaborator.User__c);
+        }
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationTriggerHelper.cls:83]: add new line

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationTriggerHelper.cls [hunk 4]
```diff
@@ -55,7 +55,47 @@ public class DEL_CaseCollaborationTriggerHelper {
...
+        Map<Id, User> map_UserById = new Map<Id, User>([SELECT 
+                                                        Id,
+                                                        IsPortalEnabled 
+                                                        FROM User 
+                                                        WHERE Id IN :set_UserIds
+                                                        ]);
+        for (DEL_CaseCollaborator__c objCaseCollaborator : list_CaseCollaborators) {
+            if(map_UserById.get(objCaseCollaborator.User__c).IsPortalEnabled){
+                objCaseCollaborator.CaseLink__c = strCommnunityURL + '/'+objCaseCollaborator.Case__c;
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationTriggerHelper.cls:92]:
```suggestion
                objCaseCollaborator.CaseLink__c = strCommnunityURL + '/' + objCaseCollaborator.Case__c;
```

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationTriggerHelper.cls [hunk 5]
```diff
@@ -55,7 +55,47 @@ public class DEL_CaseCollaborationTriggerHelper {
...
+            }
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationTriggerHelper.cls:93]:
```suggestion
            } else {
```

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationTriggerHelper.cls [hunk 6]
```diff
@@ -55,7 +55,47 @@ public class DEL_CaseCollaborationTriggerHelper {
...
+            else{
+                objCaseCollaborator.CaseLink__c = strSFURL + '/'+objCaseCollaborator.Case__c;
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationTriggerHelper.cls:95]:
```suggestion
                objCaseCollaborator.CaseLink__c = strSFURL + '/' + objCaseCollaborator.Case__c;
```

File: Delmar/force-app/main/default/triggers/DEL_CaseCollaborationTrigger.trigger [hunk 7]
```diff
@@ -24,7 +24,9 @@ trigger DEL_CaseCollaborationTrigger on DEL_CaseCollaborator__c (after insert, b
     }
 
     if (Trigger.isInsert && Trigger.isBefore) {
-        DEL_CaseCollaborationTriggerHelper.handleDuplicates(trigger.new);
+         DEL_CaseCollaborationTriggerHelper.populateFields(trigger.new);
+         DEL_CaseCollaborationTriggerHelper.handleDuplicates(trigger.new);
```
Comment (csankit0108) [inline Delmar/force-app/main/default/triggers/DEL_CaseCollaborationTrigger.trigger:28]: Call the handleDuplicates() first

**Review by csankit0108 — CHANGES_REQUESTED** [review]
@dineshdeverakonda Include the Custom field as well(CaseLink__c)

**Comment by pradeepchary** [comment]
Rename Case Link__c as Case URL (CaseUrl__c)

**Review by csankit0108 — APPROVED** [review]

**Pull request merged**